models/Event.ts

```diff
@@ -157,5 +167,61 @@ export default function (sequelize, DataTypes) {
     });
   };
 
+  /**
+   * Return the latest event of each type grouped by device id
+   */
+  Event.latestEvents = async function (user, deviceId, options) {
+    const where: any = {};
+
+    if (deviceId) {
+      where.DeviceId = deviceId;
+    }
+    const eventWhere: any = {};
+    if (options && options.eventType) {
+      if (Array.isArray(options.eventType)) {
+        eventWhere.type = {};
+        eventWhere.type[Op.in] = options.eventType;
+      } else {
+        eventWhere.type = options.eventType;
+      }
+    }
+
+    let order: any[] = [
+      ["EventDetail", "type", "DESC"],
+      ["DeviceId", "DESC"],
+      ["dateTime", "DESC"]
+    ];
+
+    return this.findAndCountAll({
```

Comment:
I think you only need to do a findAll here not a findAllAndCount.  (one less request for the sql server).